plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -1480,10 +1480,46 @@ function woocommerce_result_count() {
 		if ( ! wc_get_loop_prop( 'is_paginated' ) || ! woocommerce_products_will_display() ) {
 			return;
 		}
-		$args = array(
-			'total'    => wc_get_loop_prop( 'total' ),
-			'per_page' => wc_get_loop_prop( 'per_page' ),
-			'current'  => wc_get_loop_prop( 'current_page' ),
+
+		/**
+		 * Filters the default orderby option.
+		 *
+		 * @since 1.6.4
+		 *
+		 * @param string  $default_orderby The default orderby option.
+		 */
+		$default_orderby = apply_filters( 'woocommerce_default_catalog_orderby', get_option( 'woocommerce_default_catalog_orderby', '' ) );
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+		$orderby = isset( $_GET['orderby'] ) ? wc_clean( wp_unslash( $_GET['orderby'] ) ) : $default_orderby;
+		// phpcs:enable WordPress.Security.NonceVerification.Recommended
+
+		// If products follow the default order this doesn't need to be informed.
+		$orderby = 'menu_order' === $orderby ? '' : $orderby;
+
+		/**
+		 * Filters ordered by messages.
+		 *
+		 * @since 9.3.0
+		 *
+		 * @param array  $orderedby_messages The list of messages per orderby key.
+		 */
+		$catalog_orderedby_options = apply_filters(
```

Comment:
Same goes for this, please can we type check this result before using it. There's no guarantee this will be an `array` once all filters have run.

Reply:
@tjcafferkey Please take a look at this statement below:

```php
$orderedby = isset( $catalog_orderedby_options[ $orderby ] ) ? $catalog_orderedby_options[ $orderby ] : '';
```

If `$catalog_orderedby_options` is not an array, the result of the `isset` function will be `false` and `$orderedby` will be set to `''`. Unless I missed something, `isset` is already checking if `$catalog_orderedby_options` is an array and if `$orderby` property exists.

Reply:
Thanks for the reply @Manussakis. I agree with your explanation, I feel like we have enough safety with this check.